packages/transport/src/modules/messages/MessageController.ts

```diff
@@ -204,6 +204,34 @@ export class MessageController extends TransportController {
         return message;
     }
 
+    public async markMessageAsRead(id: CoreId): Promise<Message> {
+        const messageDoc = await this.messages.read(id.toString());
+        if (!messageDoc) {
+            throw CoreErrors.general.recordNotFound(Message, id.toString());
+        }
+
+        const message = Message.from(messageDoc);
+        if (typeof message.wasReadAt === "undefined") {
+            message.wasReadAt = CoreDate.utc();
+            await this.messages.update(messageDoc, message);
+        }
```

Comment:
please use early return where possible:

```suggestion
        if (typeof message.wasReadAt !== "undefined") return message;
        
        message.wasReadAt = CoreDate.utc();
        await this.messages.update(messageDoc, message);
```

Reply:
this was marked as resolved, but not resolved